pandas/lib.pyx

```diff
@@ -26,9 +26,9 @@ from cpython cimport (PyDict_New, PyDict_GetItem, PyDict_SetItem,
                       PyBytes_GET_SIZE,
                       PyUnicode_GET_SIZE)
 
-try:
+IF PY_MAJOR_VERSION == 2:
```

Comment:
this still seems very odd, the why can't you do

#if PY_VERSION_HEX >= 0x03000000
    from cpython cimport PyUnicode_GET_SIZE as PyString_GET_SIZE
#else
    from cpython cimport PyString_GET_SIZE
#endif
```